components/cam/src/physics/cam/shoc_intr.F90

```diff
@@ -0,0 +1,1258 @@
+    !  DIAGNOSE THE PBL DEPTH                                                           !
+    !  this is needed for aerosol code                                                  !
+    ! --------------------------------------------------------------------------------- ! 
+
+    do i=1,ncol
+      do k=1,pver
+        th(i,k) = state1%t(i,k)*state1%exner(i,k)
+        thv(i,k) = th(i,k)*(1.0_r8+zvir*state1%q(i,k,ixq))
+      enddo
+    enddo
+ 
+    ! diagnose surface friction and obukhov length (inputs to diagnose PBL depth)
+    do i=1,ncol
+      rrho(i,1) = (1._r8/gravit)*(state1%pdel(i,pver)/dz_g(i,pver))
+      call calc_ustar( state1%t(i,pver), state1%pmid(i,pver), cam_in%wsx(i), cam_in%wsy(i), &
+                       rrho(i,1), ustar2(i) )
+      call calc_obklen( th(i,pver), thv(i,pver), cam_in%cflx(i,1), cam_in%shf(i), rrho(i,1), ustar2(i), &
+                        kinheat(i), kinwat(i), kbfs(i), obklen(i) )  
+    enddo
+   
+    dummy2(:) = 0._r8
+    dummy3(:) = 0._r8
+   
+    where (kbfs .eq. -0.0_r8) kbfs = 0.0_r8
+
+    !  Compute PBL depth according to Holtslag-Boville Scheme
+    call pblintd(ncol, thv, state1%zm, state1%u, state1%v, &
+                ustar2, obklen, kbfs, pblh, dummy2, &
+                state1%zi, cloud_frac(:,1:pver), 1._r8-cam_in%landfrac, dummy3)  
+		
+    ! Assign the first pver levels of cloud_frac back to cld
+    cld(:,1:pver) = cloud_frac(:,1:pver)	
+
+    ! --------------------------------------------------------!
+    ! Output fields
+    !---------------------------------------------------------!
+
+    do k=1,pverp
+      do i=1,ncol
+        wthl_output(i,k) = wthl_sec(i,k) * rrho_i(i,pverp-k+1) * cpair
+        wqw_output(i,k) = wqw_sec(i,k) * rrho_i(i,pverp-k+1) * latvap 
+      enddo
+    enddo
+
+    do k=1,pver
+      do i=1,ncol
+        wthv_output(i,k) = wthv(i,k) * rrho(i,pver-k+1) * cpair
+        wql_output(i,k) = wqls(i,k) * rrho(i,pver-k+1) * latvap 
+      enddo
+    enddo
+
+    call outfld('SHOC_TKE', tke, pcols, lchnk)
+    call outfld('WTHV_SEC', wthv_output, pcols, lchnk)
+    call outfld('SHOC_MIX', shoc_mix, pcols, lchnk)
+    call outfld('TK', tk, pcols, lchnk)
+    call outfld('TKH', tkh, pcols, lchnk)
+    call outfld('W_SEC', w_sec, pcols, lchnk)
+    call outfld('THL_SEC', thl_sec, pcols, lchnk)
+    call outfld('QW_SEC', qw_sec, pcols, lchnk)
+    call outfld('QWTHL_SEC', qwthl_sec, pcols, lchnk)
+    call outfld('WTHL_SEC', wthl_output, pcols, lchnk)
+    call outfld('WQW_SEC', wqw_output, pcols, lchnk)
+    call outfld('WTKE_SEC', wtke_sec, pcols, lchnk)
+    call outfld('UW_SEC', uw_sec, pcols, lchnk)
+    call outfld('VW_SEC', vw_sec, pcols, lchnk)
+    call outfld('W3', w3, pcols, lchnk)
+    call outfld('WQL_SEC',wql_output, pcols, lchnk)
+    call outfld('ISOTROPY',isotropy, pcols,lchnk)
+    call outfld('CONCLD',concld,pcols,lchnk)
+    call outfld('BRUNT',brunt,pcols,lchnk)
+
+	 
+    return
+#endif             
```

Comment:
Should this be moved to line 1253 so that the "return" call is issued even if not defined SHOC_SGS?

Reply:
will fix